### PR TITLE
GPU transfer buffer synchronization control.

### PR DESCRIPTION
**Pull request by Sunwoo-Nam**
1. Remove GpuChannelMsg_RequestSyncTransferBuffer and
   GPU determines data sync timing.
2. Add the command to control synchronization for result data.

### CODE REVIEW

File: gpu/command_buffer/service/common_decoder.cc [hunk 1]
```diff
@@ -359,7 +369,29 @@ error::Error CommonDecoder::HandleGetBucketData(uint32_t immediate_data_size,
...
+#if defined(CASTANETS)
+  scoped_refptr<gpu::Buffer> buffer =
+      command_buffer_service_->GetTransferBuffer(args.shared_memory_id);
+  mojo::SyncSharedMemoryHandle(buffer->backing()->GetGUID(), offset, size);
+#endif
+  return error::kNoError;
+}
+
+#if defined(CASTANETS)
+error::Error CommonDecoder::HandleGetResultData(uint32_t immediate_data_size,
```
Comment (hyowon) [inline gpu/command_buffer/service/common_decoder.cc]: How about naming this method SyncResultData()? In Castanets, we already use Sync*() kind of names for sync methods.

**Review by hyowon — COMMENTED** [review]
A crash occurs in the browser process when quitting the browser. However, it is resolved by https://github.com/Samsung/Castanets/pull/154. Thus, this PR is going to be merged after https://github.com/Samsung/Castanets/pull/154.

File: gpu/command_buffer/service/common_decoder.cc (hunk 1 above)
Comment (Sunwoo-Nam) [reply gpu/command_buffer/service/common_decoder.cc]: I agree. I updated.

**Comment by hyowon** [comment]
#153 has been merged, so let's merge this PR too!

**Review by hyowon — APPROVED** [review]
#153 has been merged, so let's merge this PR too!

**Pull request merged**